xcvm/architecture/asset-registry/global-registry.md

```diff
@@ -0,0 +1,68 @@
+# Global Asset Registry
+
+Within `XCVM` programs, assets are identified by a global identifier, which is the same within each chain. Bitcoin for example is identified by `1`.
+
+```
+AssetId ::= u64
+```
+
+The XCVM interpreter performs a look-up when executing instructions such as `Spawn` and `Transfer` to translate the local identifier to a global one. Local identifiers are chain specific, such as `address` in EVM chains. There is a two-way supported lookup:
+
+```
+AssetsById ::= AssetId -> LocalId
+IdByAssets ::= LocalId -> AssetId
+```
+
+## Querying the Registry
+
+When constructing `Call`s, we often need to convert `LocalId`s to `LocalId`s based on the `NetworkId`. Even though USDC is an erc20 token on many chains, the actual `address` is different on each chain. For this purpose, the registry also supports converting by NetworkId:
```

Comment:
So the thing is lets say you have the same erc20 tokens on two different chains. Although the token is the same token, since the chain is different, their `LocalId` which is `Address` in this case will be different. So this is a way to identify a specific token on specific chain.

Reply:
it leads many to many maps `each chain to each chain`(many to many on each chain). while `global asset id` to `this`(local)  chain asset address/id makes map one to many on each chain.